src/fsharp/tast.fs

```diff
@@ -384,7 +384,7 @@ assert (sizeof<TyparFlags> = 4)
 
 let unassignedTyparName = "?"
 
-exception UndefinedName of int * (* error func that expects identifier name *)(string -> string) * Ident * string list
+exception UndefinedName of int * (* error func that expects identifier name *)(string -> string) * Ident * string list * string option
```

Comment:
@dsyme can you remember why UndefinedName has that string list as last parameter? In nearly all usages that thing is [] and it seems the error reporting does nothing with that list. Was designed to achieve something like this PR - but never really implemented?